test/index.ts

```diff
@@ -142,22 +142,23 @@ describe('teeny', () => {
     });
   });
 
-  it('should respect environment variables for proxy config', () => {
-    const envVars = ['http_proxy', 'https_proxy', 'HTTP_PROXY', 'HTTPS_PROXY'];
-    for (const v of envVars) {
+  const envVars = ['http_proxy', 'https_proxy', 'HTTP_PROXY', 'HTTPS_PROXY'];
+  for (const v of envVars) {
+    it('should respect environment variables for proxy config', done => {
       sandbox.stub(process, 'env').value({[v]: 'https://fake.proxy'});
       const expectedBody = {hello: '🌎'};
       const scope = nock(uri)
         .get('/')
         .reply(200, expectedBody);
       teenyRequest({uri}, (err, res, body) => {
-        assert.ifError(err);
         scope.done();
+        assert.ifError(err);
         assert.deepStrictEqual(expectedBody, body);
         assert.ok(res.request.agent instanceof HttpsProxyAgent);
```

Comment:
We might want to consider breaking creating separate tests for http/https so we can test that correct agent is being used based on the env var?

Reply:
This isn't quite how the proxy libraries work, I believe both `https-proxy-agent` and `http-proxy-agent` can connect to an upstream `http` _or_ `https` proxy.

The difference is in whether the URL being connected to is ultimately `https`.

> This module provides an http.Agent implementation that connects to a specified HTTP or HTTPS proxy server, and can be used with the built-in https module.